packages/relay/src/lib/relay.ts

```diff
@@ -141,13 +149,15 @@ export class RelayImpl implements Relay {
       process.env.MIRROR_NODE_URL_WEB3 || process.env.MIRROR_NODE_URL || '',
     );
 
-    this.metricService = new MetricService(
+    // Note: Since the main capturing metric logic of the `MetricService` class works by listening to specific events,
+    //       this class does not need an instance but must still be initiated.
+    new MetricService(
       logger,
       hapiService.getSDKClient(),
       this.mirrorNodeClient,
-      hbarLimiter,
       register,
       this.eventEmitter,
+      hbarLimitService,
     );
```

Comment:
Let’s still bound it to a field, any references to the instance are getting lost this way and it might get destroyed by the garbage collector at some point.

Reply:
You are right without any references to the instance, it could indeed be garbage collected at some point. Reverted.